programs/steward/src/instructions/set_new_authority.rs

```diff
@@ -1,7 +1,82 @@
+use anchor_lang::idl::types::*;
 use anchor_lang::prelude::*;
+use anchor_lang::IdlBuild;
+use borsh::{BorshDeserialize, BorshSerialize};
 
 use crate::{errors::StewardError, state::Config};
 
+#[repr(u8)]
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq)]
+pub enum AuthorityType {
+    SetAdmin = 0,
+    SetBlacklistAuthority = 1,
+    SetParametersAuthority = 2,
+}
+
+impl AuthorityType {
+    pub fn to_u8(self) -> u8 {
+        self as u8
+    }
+
+    pub fn into(&self) -> u8 {
+        match self {
+            AuthorityType::SetAdmin => 0,
+            AuthorityType::SetBlacklistAuthority => 1,
+            AuthorityType::SetParametersAuthority => 2,
+        }
+    }
+
+    pub fn from(index: u8) -> Result<Self> {
```

Comment:
`into` and `from` aren't typically implemented as struct methods, they're their own trait methods 
`impl From<u8> for AuthorityType {}`

Reply:
Done